### PR TITLE
added visionary badge

### PR DESCRIPTION
**Pull request by pg20**

### CODE REVIEW

File: app/controllers/award_badges_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class AwardBadgesController < ApplicationController
+
+  def index
+    @projects = Project.where.not(mentee_id: Badge.where(name: [11]).includes(:users).map{|b| b.users.pluck(:id)}.flatten)
```
Comment (DMips) [inline app/controllers/award_badges_controller.rb]: filtering `mentee id` looks very complicated here. How about `Badge.visionary.joins(:users).uniq.pluck("users.id")` it will also choose mentees that already have assigned visionary badge but for me it looks simpler - what do you think?

File: app/views/award_badges/index.html.slim [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    .table-responsive
+      table.table.table-striped
+        thead
+          tr
+            th.col-md-1 #
+            th.col-md-2 Mentee
+            th.col-md-2 Project Description
+            th.col-md-6 
+        tbody
+        - if @projects != nil
```
[the remark this answers is not in the document]
Comment (DMips) [reply app/views/award_badges/index.html.slim]: Unless you want to display message like `All projects have assigned visionary badges` then the condition will be
``` 
- if @projects.any?
    iterate 
- else
  "Visionary badges awarded to all mentees" or whatever text you want to put here
```

**Comment by pg20** [comment]
@DMips I have made the changes, please review again.

File: app/views/organisers/award_badges.slim [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    .table-responsive
+      table.table.table-striped
+        thead
+          tr
+            th.col-md-1.col-md-offset-2.col-sm-1.col-sm-offset-1.col-xs-1 #
+            th.col-md-2.col-md-offset-2.col-sm-2.col-sm-offset-1.col-xs-2 Mentee
+            th.col-md-100.col-md-offset-2.col-sm-100.col-sm-offset-1.col-xs-100 Project Description
+            th.col-md-6.col-md-offset-2.col-sm-6.col-sm-offset-1.col-xs-6 
+        tbody
+        - if @projects.any?
```
Comment (DMips) [inline app/views/organisers/award_badges.slim]: that should be nested inside `tbody` - currently it's on the same level so following `tr` tags won't be children of `tbody`

File: app/controllers/organisers_controller.rb [hunk 4]
```diff
@@ -52,8 +52,26 @@ def problematic_projects
...
+    @project = Project.find(params[:id])
+    assign_visionary_badge
+    redirect_to award_badges_organisers_path
+  end
+
+
   private
 
+  def assign_visionary_badge
+      unless @project.mentee.badges.visionary.any?
```
Comment (DMips) [inline app/controllers/organisers_controller.rb]: just a note - indentation in rails is 2 spaces not 4

File: config/routes.rb [hunk 5]
```diff
@@ -13,6 +13,8 @@ def draw(routes_name)
     collection do
       get :dashboard
       get :problematic_projects
+      get :award_badges
```
Comment (DMips) [inline config/routes.rb]: why did you move it to organizers controller? can we move it back to separate controller?

File: app/views/organisers/award_badges.slim [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+= link_to "Back to All projects", dashboard_organisers_path
+.row
+  .col-xs-30
+    .table-responsive
+      table.table.table-striped
+        thead
+          tr
+            th.col-md-1.col-md-offset-2.col-sm-1.col-sm-offset-1.col-xs-1 #
+            th.col-md-2.col-md-offset-2.col-sm-2.col-sm-offset-1.col-xs-2 Mentee
+            th.col-md-100.col-md-offset-2.col-sm-100.col-sm-offset-1.col-xs-100 Project Description
```
Comment (DMips) [inline app/views/organisers/award_badges.slim]: what's `col-md-100`? and why do we need offsets here?

File: config/routes.rb (hunk 5 above)
Comment (pg20) [reply config/routes.rb]: Well I moved it to organiser controller because the , Controller was just doing one functionality. Also it was organiser task to award badge, to so I moved the award_function here. 
Also, only organisers should be able to see this page, so I thought of moving everything.

Comment (DMips) [reply config/routes.rb]: @pg20 I think it's too much logic in one controller now - awarding badges is separate thing and can be easily separated (just like you did before) so I'd be for moving it back to separate controller. Also I sent you line of code few days back (I think you asked for it on slack) which checks whether logged user is organizer to allow only organizer to access page

Comment (pg20) [reply config/routes.rb]: Yes you gave the link of organiser controller, but i misinterpreted. So I added here :(. Ill move everything back by today.

**Comment by pg20** [comment]
@DMips I have done the changes as discussed. Also I made changes in user_session controller as discussed. I am not able to make project description wide enough, if you can make that changes while merging. I have added award_badges link in ongoing_projects page in organiser_view. It can only be seen once logged in as organiser now.
